src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/UserTimeZoneService.cs

```diff
@@ -22,17 +26,17 @@ public class UserTimeZoneService
     public UserTimeZoneService(
         IClock clock,
         IDistributedCache distributedCache,
-        IHttpContextAccessor httpContextAccessor
-    )
+        IHttpContextAccessor httpContextAccessor)
     {
         _clock = clock;
         _distributedCache = distributedCache;
         _httpContextAccessor = httpContextAccessor;
     }
 
-    public async ValueTask<ITimeZone> GetUserTimeZoneAsync()
+    /// <inheritdoc/>
+    public async ValueTask<ITimeZone> GetAsync(IUser user)
     {
-        var currentTimeZoneId = await GetCurrentUserTimeZoneIdAsync();
+        var currentTimeZoneId = await GetTimeZoneIdAsync();
```

Comment:
You cannot get the time zone from the current user. You'll need the time zone of the given using. Pass `user.UserName` to the private method to get the time zone of the given user

Reply:
Or we can pass user instead because we need it in there